pkg/cluster/cluster.go

```diff
@@ -205,17 +205,18 @@ func (c *Cluster) send(ev *clusterEvent) {
 }
 
 func (c *Cluster) run(stopC <-chan struct{}, wg *sync.WaitGroup) {
-	needDeleteCluster := true
+	clusterFailed := false
 
 	defer func() {
-		if needDeleteCluster {
-			c.logger.Infof("deleting cluster")
+		if clusterFailed {
+			c.reportFailedStatus()
```

Comment:
why cluster will block the controller? i do not think it ever should. if cluster blocks controller, then there is something wrong somewhere else i guess.